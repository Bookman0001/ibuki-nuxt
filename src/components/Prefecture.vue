<template>
  <div
    :class="{
      'safe-zone prefecture-item': prefectureInfection.cases < 700,
      'caution-zone prefecture-item': prefectureInfection.cases >= 700,
      'danger-zone prefecture-item': prefectureInfection.cases >= 2000,
      'most-danger-zone prefecture-item': prefectureInfection.cases >= 5000,
    }"
  >
    <div class="wrapper">
      <span>{{ prefectureInfection.nameJa }}</span>
    </div>
    <div class="wrapper">
      <label class="label-item">これまでの感染者数</label>
      <span>{{ prefectureInfection.cases }}人</span>
    </div>
    <div class="wrapper">
      <label class="label-item">これまでのPCR検査数</label>
      <span>{{ prefectureInfection.pcr }}人</span>
    </div>
    <div class="wrapper">
      <label class="label-item">これまでの死者数</label>
      <span>{{ prefectureInfection.deaths }}人</span>
    </div>
    <div class="wrapper">
      <label class="label-item">入院患者数</label>
      <span>{{ prefectureInfection.hospitalize }}人</span>
    </div>
    <div class="wrapper">
      <label class="label-item">重症者数</label>
      <span>{{ prefectureInfection.severe }}人</span>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType } from '@nuxtjs/composition-api'
import { Infection } from '~/types'

export default defineComponent({
  props: {
    prefectureInfection: {
      type: Object as PropType<Infection>,
      default: () => ({
        id: 0,
        nameJa: '',
        nameEn: '',
        lat: '',
        lng: '',
        pct: 0,
        population: 0,
        cases: 0,
        deaths: 0,
      }),
    },
  },
})
</script>

<style scoped>
.prefecture-item {
  font-size: 1rem;
  font-weight: 500;
  border-radius: 10px;
  border: 1px solid #303030;
  padding: 10px;
  margin: 10px;
}
.safe-zone {
  background-color: #26499d;
  color: #fff;
}
.caution-zone {
  background-color: #ff9;
}
.danger-zone {
  background-color: #d70035;
  color: #fff;
}
.most-danger-zone {
  background-color: #a50082;
  color: #fff;
}
.wrapper {
  width: 100%;
}
.label-item {
  margin: 5px;
}
</style>
