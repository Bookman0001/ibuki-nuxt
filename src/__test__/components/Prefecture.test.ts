import { mount } from '@vue/test-utils'
import Prefecture from '~/components/Prefecture.vue'

describe('Prefecture', () => {
  test('render correctly', () => {
    const prefecture = mount(Prefecture, {
      propsData: {
        id: 1,
        nameJa: 'テスト都道府県',
        nameEn: 'test-prefecture',
        lat: 0,
        lng: 0,
        population: 10000,
        cases: 10,
        deaths: 0,
        pcr: 1000,
        hospitalize: 0,
        severe: 0,
      },
    })
    expect(prefecture.element).toMatchSnapshot()
  })
})
