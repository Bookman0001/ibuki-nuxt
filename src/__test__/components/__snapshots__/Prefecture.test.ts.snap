// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prefecture exist component 1`] = `
<div
  cases="10"
  class="safe-zone prefecture-item"
  deaths="0"
  hospitalize="0"
  id="1"
  lat="0"
  lng="0"
  nameen="test-prefecture"
  nameja="テスト都道府県"
  pcr="1000"
  population="10000"
  severe="0"
>
  <div
    class="wrapper"
  >
    <span>
      
    </span>
  </div>
   
  <div
    class="wrapper"
  >
    <label
      class="label-item"
    >
      これまでの感染者数
    </label>
     
    <span>
      0人
    </span>
  </div>
   
  <div
    class="wrapper"
  >
    <label
      class="label-item"
    >
      これまでの死者数
    </label>
     
    <span>
      0人
    </span>
  </div>
   
  <div
    class="wrapper"
  >
    <label
      class="label-item"
    >
      入院患者数
    </label>
     
    <span>
      人
    </span>
  </div>
   
  <div
    class="wrapper"
  >
    <label
      class="label-item"
    >
      重症者数
    </label>
     
    <span>
      人
    </span>
  </div>
</div>
`;

exports[`Prefecture render correctly 1`] = `
<div
  cases="10"
  class="safe-zone prefecture-item"
  deaths="0"
  hospitalize="0"
  id="1"
  lat="0"
  lng="0"
  nameen="test-prefecture"
  nameja="テスト都道府県"
  pcr="1000"
  population="10000"
  severe="0"
>
  <div
    class="wrapper"
  >
    <span>
      
    </span>
  </div>
   
  <div
    class="wrapper"
  >
    <label
      class="label-item"
    >
      これまでの感染者数
    </label>
     
    <span>
      0人
    </span>
  </div>
   
  <div
    class="wrapper"
  >
    <label
      class="label-item"
    >
      これまでの死者数
    </label>
     
    <span>
      0人
    </span>
  </div>
   
  <div
    class="wrapper"
  >
    <label
      class="label-item"
    >
      入院患者数
    </label>
     
    <span>
      人
    </span>
  </div>
   
  <div
    class="wrapper"
  >
    <label
      class="label-item"
    >
      重症者数
    </label>
     
    <span>
      人
    </span>
  </div>
</div>
`;
